app.py

```diff
@@ -82,4 +92,62 @@ def getUsernamePasswordFromHeader(req) -> Tuple[str, str]:
         return "",""
     hashBytes = req.headers["Authorization"].split()[1]
     username, passsword = base64.b64decode(hashBytes).decode("utf-8").split(":")
-    return username, passsword
\ No newline at end of file
+    return username, passsword
+
+
+@app.route("/startgame", methods=["GET", "POST"])
+async def startGame ():
+    if request.method == "GET":
+        return "Pass in username to start the game"
+    else:
+        data = await request.get_json()
+        if not data or 'username' not in data:
+            return "Required username", 400
+        isUser = await isUserExisted(data['username'])
+        if not isUser:
+            return "Username not Found, Please register " + data['username'], 404
+        db =  get_db()
+        cur = db.execute(
+            "SELECT username FROM games WHERE username = ?", (data["username"],)
+        )
+        user = cur.fetchall()
+        if user:
+            return "Game Exists"
+        f = open('correct.json')
+        jsonData = json.load(f)
+        try:
+            id =  db.execute(
+            "INSERT INTO games(username, secretkey,numberOfGuesses) VALUES(?, ?, ?)", (data["username"], random.choice(jsonData),0,)
+            )
+            db.commit()
+        except sqlite3.IntegrityError as e:
+            abort(409, e)
+        return f'game started with id: {id.lastrowid}' 
+
+@app.route("/listallgames/<string:username>", methods=["GET"])
+def listAllGame (username):
+    db =  get_db()
+    game =  db.execute(
+            "SELECT gameId,numberOfGuesses, username FROM games WHERE username = ?", (username,)
+        )
+
+    game = game.fetchall()
+    print("game",game)
+    if game:
+        return list(map(dict, game))
+    else:
+        abort(404)
```

Comment:
Maybe we can do a message like "No record" instead of standard 404 so there is more feedback?
Since 404 alone maybe too vague, other things could happen as well

Reply:
I have taken this from the profs code , still ill work on this later